### DISCUSSION TITLE
TagAction does not support subtype Ruin

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14394] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

TagAction does not have Ruin (and EnemySub!) as a valid submarine type for some reason, so you can't tag things inside of ruins for events

https://github.com/FakeFishGames/Barotrauma/blob/0aee53976202c7ea5c0bb3f26669a97c8dc2400c/Barotrauma/BarotraumaShared/SharedSource/Events/EventActions/TagAction.cs#L14

![image](https://github.com/user-attachments/assets/af9a8ff4-3291-40ff-b958-8ae0dc1eeb00)

### Reproduction steps

```xml
    <ScriptedEvent identifier="canttagruins" commonness="100">
      <TagAction criteria="player" tag="player" ContinueIfNoTargetsFound="true" />
      <TagAction criteria="hull" tag="ruin" submarinetype="ruin" ContinueIfNoTargetsFound="false" />
      <ConversationAction text="You tagged a ruin" dialogtype="Small" targettag="player">
    </ScriptedEvent>
```

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
This has been addressed in the latest unstable build (see https://github.com/FakeFishGames/Barotrauma/pull/13974).
